src/ipc/handler.c

```diff
@@ -185,13 +193,6 @@ static int ipc_stream_pcm_params(uint32_t stream)
 
 	trace_ipc("ipc: comp %d -> params", pcm_params->comp_id);
 
-	/* sanity check size */
-	if (IPC_INVALID_SIZE(pcm_params)) {
-		trace_ipc_error("ipc:_invalid IPC size 0x%x got 0x%x",
-				sizeof(*pcm_params), pcm_params->hdr.size);
-		return -EINVAL;
-	}
```

Comment:
Not sure I see what the size check has to do with the ABI?
It's still useful not to trust the kernel, isn't it? Even for development it helps identify mismatches.

Reply:
I think this has to work both ways since the driver also has to trust the FW too. We have 2 situations (for MINOR, PATCH updates) we need to deal with :

1) Sender sends larger IPC than receiver expects - Easy to deal with as receiver only uses what it knows and by default ignores any new data.
2) Sender sends smaller IPC than receiver expects - More complex, but receiver should pad any missing data as 0 i.e. all MINOR/PATCH ABI additions require non zero data to activate.

Reply:
Agree, but we are doing neither. The patch just removes all checks and does not add padding or overrides.

Reply:
Fixing that now but will be new PR. Will need something similar in the kernel.

Reply:
ah ok then, didn't realize there was a planned follow-up